elasticsearch-transport/lib/elasticsearch/transport/transport/sniffer.rb

```diff
@@ -8,14 +8,15 @@ class Sniffer
         ES1_RE_URL  = /\[([^\/]*)?\/?([^:]*):([0-9]+)\]/
         ES2_RE_URL  = /([^\/]*)?\/?([^:]*):([0-9]+)/
 
-        attr_reader   :transport
+        attr_reader   :transport, :logger
         attr_accessor :timeout
 
         # @param transport [Object] A transport instance
         #
-        def initialize(transport)
+        def initialize(transport, logger=nil)
```

Comment:
Can we add a tiny unit test here? Even something super tiny such as `should "be initialized with a logger instance" do ...`.